### PR TITLE
FYLE-1kejn9: Initial API impl

### PR DESCRIPTION
**Pull request by abhishekk728**

### CODE REVIEW

File: fylesdk/internals/network.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            url (str): Url for the wanted API.
+
+        Returns:
+            A response from the request (dict).
+        """
+        return self._delete(url, **kwargs)
+
+
+    def _assert_response(self, response):
+        if response.status_code == 400:
```
Comment (jayachandhar) [inline fylesdk/internals/network.py:67]: response should have valid json with proper error message

File: fylesdk/fyle_platform_sdk.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    def update_access_token(self):
+        """Update the access token."""
+
+        self.__get_access_token()
+    
+
+    def set_server_url(self):
+        """Set the Base URL in all API objects."""
+
+        sdk.config['SERVER_URL'] = self.__base_url
```
Comment (jayachandhar) [inline fylesdk/fyle_platform_sdk.py:42]: can we use different base url for token and getting data ? 
eg: token url: accounts.fylehq.com/router/token
server url: platform.fylehq.com/v1/admin/employees

File: fylesdk/apis/v1/fyler/expense_comments.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+from .role import role
+from ..version import version
+from ...api_base import ApiBase
+
+
+class ExpenseComments(ApiBase):
+    """Class for Expense Comments APIs."""
+
+    POST_EXPENSE_COMMENT = '/expense_comment'
```
Comment (jayachandhar) [inline fylesdk/apis/v1/fyler/expense_comments.py:15]: expense_comment -> expense_comments

File: fylesdk/internals/network.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+import requests
+
+from . import serializers
+from .. import exceptions
+
+class Network:
+    """Class for making GET, POST requests"""
+    HOSTNAME = None
+    HEADERS = serializers.deserialize({
+            'USER_DETAILS': 'X-User-Details',
```
[the remark this answers is not in the document]
Comment (abhishekk728) [reply fylesdk/internals/network.py:16]: ok will remove this, but this is also present in fyle-sdk

File: fylesdk/globals/configs.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+"""
```
[the remark this answers is not in the document]
Comment (abhishekk728) [reply fylesdk/globals/configs.py:1]: based on below defs, i made in separate packages:

globals -> stuffs which can be changed, updated , used from everywhere
internals -> stuffs which are only for particular usages

no other strong points.

File: fylesdk/internals/network.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    def post_request(self, url, **kwargs):
+        """Create a HTTP post request.
+
+        Parameters:
+            url (str): Url for the wanted API.
+
+        Returns:
+            A response from the request (dict).
+        """
+        return self._post(url, **kwargs)
```
[the remark this answers is not in the document]
Comment (abhishekk728) [reply fylesdk/internals/network.py:51]: removing this doesn't providing any value.

File: fylesdk/internals/network.py [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+        """Create a HTTP GET request.
+
+        Parameters:
+            url (str): Url for the wanted API.
+
+        Returns:
+            A response from the request (dict).
+        """
+
+        return self._get(url, **kwargs)
```
[the remark this answers is not in the document]
Comment (abhishekk728) [reply fylesdk/internals/network.py:39]: removing this doesn't providing any value.

File: setup.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    license='MIT',
+    long_description=long_description,
+    long_description_content_type='text/markdown',
+    keywords=['fyle', 'api', 'python', 'sdk'],
+    url='https://github.com/fylein/fyle-platform-sdk-py',
+    packages=setuptools.find_packages(),
+    install_requires=[
+        'enum34==1.1.10',
+        'requests==2.24.0',
+        'Flask==1.1.2',
```
[the remark this answers is not in the document]
Comment (abhishekk728) [reply setup.py]: replaced with configparser its a part of cpython

File: fylesdk/internals/network.py (hunk 1 above)
Comment (abhishekk728) [reply fylesdk/internals/network.py]: updated

File: fylesdk/fyle_platform_sdk.py (hunk 2 above)
Comment (abhishekk728) [reply fylesdk/fyle_platform_sdk.py]: have TOKEN_URL and SERVER_URL now

File: fylesdk/apis/v1/fyler/expense_comments.py (hunk 3 above)
Comment (abhishekk728) [reply fylesdk/apis/v1/fyler/expense_comments.py]: updated

File: fylesdk/apis/v1/admin/role.py [hunk 9]
```diff
@@ -0,0 +1 @@
+role = 'admin'
```
Comment (jayachandhar) [inline fylesdk/apis/v1/admin/role.py:1]: can this not specified in fylesdk/apis/v1/admin/__init__.py ?

**Pull request closed without merging**